lib/strings.nix

```
@@ -1424,6 +1424,38 @@ rec {
   */
   toUpper = replaceStrings lowerChars upperChars;
 
+  /**
+    Converts the first character of a string `s` to upper-case.
+
+    # Inputs
+
+    `str`
+    : The string to convert to sentence case.
+
+    # Type
+
+    ```
+    toSentenceCase :: string -> string
+    ```
+
+    # Examples
+    :::{.example}
+    ## `lib.strings.toSentenceCase` usage example
+
+    ```nix
+    toSentenceCase "home"
+    => "Home"
+    ```
+
+    :::
+  */
+  toSentenceCase = str:
+    let
+      firstChar = substring 0 1 str;
+      rest = substring 1 (stringLength str) str;
+    in
+    toUpper firstChar + toLower rest;
```

I might be wrong but i think substring doesn't propagate the string context. So if not we should propagate it manually. 
```suggestion
    addContextFrom str (toUpper firstChar + toLower rest);
```